pilot/pkg/serviceregistry/kube/controller/ambient/waypoints.go

```diff
@@ -146,7 +146,9 @@ func fetchWaypointForWorkload(ctx krt.HandlerContext, Waypoints krt.Collection[W
 // defaultNamespace avoids the need to infer when object meta from a namespace was given
 func getUseWaypoint(meta metav1.ObjectMeta, defaultNamespace string) (named *krt.Named, isNone bool) {
 	if labelValue, ok := meta.Labels[constants.AmbientUseWaypointLabel]; ok {
-		if labelValue == "#none" || labelValue == "~" {
+		// NOTE: this means Istio reserves the word "none" in this field with a special meaning
+		//   a waypoint named "none" cannot be used and will be ignored
```

Comment:
To be honest I liked the idea behind "#none" and "-"  

I expect we'll need other values to indicate non-Istio gateways or gateways in a different system. The same applies to the 'data plane mode'.

Not against this PR - we don't use "#" in other places - but I suggest adopting the naming using by Sidecar.egress and VirtualServices.gateways, with ~ to indicate no sidecar ( and in the future NAMESPACE/NAME to indicate a waypoint in a different namespace - and other extensions based on "path label" and presence of "/" separator ). 

"~" is used in Sidecar and I think we should continue to support it here as well ( even if we keep none too ).

Reply:
actually this brings up an interesting point. A user likely could have a waypoint called `none` if we allow `namespace/name` format but they will not be able to use the short form. They could use `my-cool-namespace/none` to get around the limitation of reserving `none`.

That being said something like `~` isn't a bad shout either and I'm happy to put it back in. I had removed it because it wasn't documented but I'm not hard opposed to it.

Reply:
Not sure if you saw discussion on #ambient-dev @costinm, a few folks thought #none was a typo in our doc.  Given it is not consistent with anywhere else in Istio, we don't see why we just reserve `none` as a reserved name.

supporting `~` is okay if you feel very strongly about it but I don't want to doc it given it is not documented by sidecars.

Reply:
I did add back ~ and I also went ahead and excluded a suffix of "/none" to prevent attempts to work around the reserved wording by the example I had provided above with namespace/none

Reply:
I'm up for documenting it in both places